includes/helper.php

```diff
@@ -0,0 +1,418 @@
+                    case 'MultiPoint':
+                        // Convert MultiPoint to Point by taking the first coordinate
+                        if (isset($geometry['coordinates'][0])) {
+                            $geometry['coordinates'] = $geometry['coordinates'][0];
+                            $geometry['type'] = 'Point';
+                        }
+                        break;
+                    case 'MultiLineString':
+                        // Convert MultiLineString to LineString by taking the first set of coordinates
+                        if (isset($geometry['coordinates'][0])) {
+                            $geometry['coordinates'] = $geometry['coordinates'][0];
+                            $geometry['type'] = 'LineString';
+                        }
+                        break;
+                    case 'MultiPolygon':
+                        // Convert MultiPolygon to Polygon by taking the first polygon's coordinates
+                        if (isset($geometry['coordinates'][0])) {
+                            $geometry['coordinates'] = $geometry['coordinates'][0];
+                            $geometry['type'] = 'Polygon';
+                        }
+                        break;
+                }
+            }
+
+            // Construct GeoJSON Feature
+            $feature = [
+                'type' => 'Feature',
+                'geometry' => $geometry,
+                'properties' => $properties,
+            ];
+
+            $geoJson['features'][] = $feature;
+
+        }
+
+        // Pretty‐print & unescape slashes/unicode for readability
+        return json_encode(
+            $geoJson ,
+            JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE
+        );
+
+    }
+}
+
+if ( !function_exists( 'draad_maps_populate_infowindow' ) ) {
+    /**
+     * Populate dataset content repeater with keys from feature properties
+     */
+    function draad_maps_populate_infowindow( $post_id ) {
+
+        $post = get_post( $post_id );
+
+        if ( !$post ) {
+            error_log( 'Draad Kaarten | draad_maps_populate_infowindow() - no post found' );
+            return;
+        }
+
+        setup_postdata( $post );
+
+        $datasets = get_field( 'datasets' );
+        if ( !is_iterable( $datasets ) || empty( $datasets ) ) {
+            error_log( 'Draad Kaarten | draad_maps_populate_infowindow() - no datasets found' );
+            return;
+        }
+
+        $newFieldValue = [];
+        foreach ( $datasets as $index => $dataset ) {
+            
+            $newFieldValue[] = $dataset;
+
+            $endpoint = $dataset['type_dataset'] === 'api' ? $dataset['api_endpoint'] : ( $dataset['type_dataset'] === 'file' ? wp_get_attachment_url( $dataset['file'] ) : '' );
+            $data = draad_maps_get_data( $endpoint );
+
+            if ( !$data ) {
+                continue;
+            }
+
+            $data = json_decode( $data );
+
```

Comment:
`json_decode` geeft geen errors, je moet daarvoor de method `json_last_error()` gebruiken (https://www.php.net/manual/en/function.json-last-error.php). Als die `JSON_ERROR_NONE` is, dan weet je dat de `json_decode` succesvol was, en dat `$data` in ieder geval geldige JSON is.